opentelemetry-api/src/opentelemetry/metrics/__init__.py

```diff
@@ -0,0 +1,384 @@
+# Copyright The OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# pylint: disable=too-many-ancestors
+# type: ignore
+
+# FIXME enhance the documentation of this module
+"""
+This module provides abstract and concrete (but noop) classes that can be used
+to generate metrics.
+"""
+
+
+from abc import ABC, abstractmethod
+from logging import getLogger
+from os import environ
+from typing import Optional, cast
+
+from opentelemetry.environment_variables import OTEL_PYTHON_METER_PROVIDER
+from opentelemetry.metrics.instrument import (
+    Counter,
+    DefaultCounter,
+    DefaultHistogram,
+    DefaultObservableCounter,
+    DefaultObservableGauge,
+    DefaultObservableUpDownCounter,
+    DefaultUpDownCounter,
+    Histogram,
+    ObservableCounter,
+    ObservableGauge,
+    ObservableUpDownCounter,
+    UpDownCounter,
+)
+from opentelemetry.util._providers import _load_provider
+
+_logger = getLogger(__name__)
+
+
+class MeterProvider(ABC):
```

Comment:
Rebase?

Reply:
yes, sorry, it should be a matter of rebasing this PR against `metrics_new`